outline/shadowsocks/config.go

```diff
@@ -35,16 +42,47 @@ type Client struct {
 	client.Client
 }
 
+func extractPrefixBytes(prefixUtf8 string) ([]byte, error) {
+	// The prefix is an 8-bit-clean byte sequence, stored in the codepoint
+	// values of a unicode string, which arrives here encoded in UTF-8.
+	prefixRunes := []rune(prefixUtf8)
+	prefixBytes := make([]byte, len(prefixRunes))
+	for i, r := range prefixRunes {
+		if (r & 0xFF) != r {
+			return nil, fmt.Errorf("character out of range: %d", r)
+		}
+		prefixBytes[i] = byte(r)
+	}
+	return prefixBytes, nil
+}
+
 // NewClient provides a gobind-compatible wrapper for [client.NewClient].
 func NewClient(config *Config) (*Client, error) {
-	c, err := client.NewClient(config.Host, config.Port, config.Password, config.CipherName)
+	c, err := client.NewClient(config.Host, config.Port, config.Password, config.Method)
 	if err != nil {
 		return nil, err
 	}
 	if len(config.Prefix) > 0 {
 		log.Debugf("Using salt prefix: %s", string(config.Prefix))
-		c.SetTCPSaltGenerator(client.NewPrefixSaltGenerator(config.Prefix))
+		prefixBytes, err := extractPrefixBytes(config.Prefix)
+		if err != nil {
+			return nil, fmt.Errorf("prefix parsing failed: %w", err)
+		}
+		c.SetTCPSaltGenerator(client.NewPrefixSaltGenerator(prefixBytes))
 	}
 
 	return &Client{c}, nil
 }
+
+// NewConfig converts a UTF-8 JSON string into a Config struct.
+// A string is used instead of
+func NewConfig(s string) (*Config, error) {
+	var config Config
+	if err := json.Unmarshal([]byte(s), &config); err != nil {
+		return nil, err
+	}
+	if len(config.Host) == 0 || config.Port == 0 || len(config.Method) == 0 || len(config.Password) == 0 {
```

Comment:
```suggestion
	if len(config.Host) == 0 || config.Port <= 0 || config.Port > 65535 || len(config.Method) == 0 || len(config.Password) == 0 {
```

Reply:
I've changed `Port` to a uint16 in the JSON structure, and confirmed via the unit tests that `json.Unmarshal` provides the appropriate range check.